app/__main__.py

```diff
@@ -0,0 +1,107 @@
+import abc
+import csv
+import datetime
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union, Dict, List, Optional, Generator
+
+import dateutil
+import dateutil.parser
+
+
+@dataclass
+class Person:
+    address: str
+    data: Dict[str, Union[str, datetime.datetime]]
+
+
+class BaseResource(abc.ABC):
+
+    @abc.abstractmethod
+    def connect(self):
+        ...
+
+    @abc.abstractmethod
+    def parse(self):
+        ...
+
+
+class CSVResource(BaseResource):
+    def __init__(self, path):
+        self._path = path
+        self._reader: Optional[List[Dict[str, Union[str, datetime]]]] = None
+
+    def connect(self) -> None:
+        with open(self._path, mode='r') as csv_input:
+            self._reader = list(csv.DictReader(csv_input))
+
+    def parse(self) -> Generator:
+        assert self._reader
+        for person in self._reader:
+            yield Person(address=person[' email'], data=person)
+
+
+class Handler:
+    def __init__(self, resource: BaseResource):
+        self._resource = resource
+
+    def _select_birthday_person(self, person_data):
+        person_birthday = dateutil.parser.parse(person_data[" date_of_birth"])
+
+        if int(person_birthday.month) == datetime.datetime.today().month == 2 \
```

Comment:
Few things regarding the IF statements you have here.

One: inconsistency
Sometimes you check `int(person_birthday.month) == 2` the other time you do `person_birthday.day == '29'`. Try to be consistent as there should be one and only way to do it.

Two: complex IF's
Your IF's are not really readable. You can try to split and give them a meaningful names.
```
today = datetime.date.today()
born_month = int(person_birthday.month)
born_day = int(person_birthday.day) 
born_on_february_29 =  born_month == 2 and born_day == 29
today_is_february_28 = today.day == 28
born_today = today.day == born_day and today.month == born_month

if born_today: 
    return True
elif (born_on_february_29 and today_is_february_28): 
    return True
return False
```

It should make the logic more readable.

Reply:
yeah, you are right! this piece of code left untouched from Dojo :) and during meetup this code was couple of times refactored and left with some mess of strings/ints comparing.